node/src/command.rs

```diff
@@ -504,22 +504,29 @@ pub fn run() -> Result<()> {
 			let chain_spec = &runner.config().chain_spec;
 
 			set_default_ss58_version(chain_spec);
+
+			use sc_executor::{sp_wasm_interface::ExtendedHostFunctions, NativeExecutionDispatch};
+			type HostFunctionsOf<E> = ExtendedHostFunctions<
+				sp_io::SubstrateHostFunctions,
+				<E as NativeExecutionDispatch>::ExtendHostFunctions,
+			>;
+
 			// grab the task manager.
 			let registry = &runner.config().prometheus_config.as_ref().map(|cfg| &cfg.registry);
 			let task_manager = TaskManager::new(runner.config().tokio_handle.clone(), *registry)
 				.map_err(|e| format!("Error: {:?}", e))?;
 
 			if chain_spec.is_crab() {
 				runner.async_run(|config| {
-					Ok((cmd.run::<Block, CrabRuntimeExecutor>(config), task_manager))
+					Ok((cmd.run::<Block, HostFunctionsOf<CrabRuntimeExecutor>>(), task_manager))
```

Comment:
`runner.async_run(|config| {...}` => `runner.async_run(|_| {...}`?

Reply:
Please do not forget to update the `on_runtime_upgrade` function in the runtime side.